### DISCUSSION TITLE
Conditional Windows Code

### DISCUSSION THREAD

**Discussion by Edgarianom1212** [AvaloniaUI/Avalonia#17372] [Q&A] [answered]
Is there a way to include e.g. Excel in an Avalonia project if the current OS is Windows?

**Comment by thevortexcloud** [comment, marked as answer]
There are several ways. If you just want to do a conditional assembly reference, the easiest option is to target the Windows specific TFM (EG `.net8.0-windows`) and then do a conditional reference.

**Reply by Edgarianom1212** [reply]
Thanks, I will look into that!

**Reply by thevortexcloud** [reply]
You are welcome. If you want a more portable solution (as aforementioned approach wont work on Linux as there is no Linux specific TFM), you can also define custom compile constants (EG `WINDOWS` or `LINUX`) and do a similar thing. You can also do a less portable approach by making the condition based on the build PCs operating system. But that's not great for people who want to cross compile/for CI systems.

There are many other ways to do this, but those are the most common.
